app/views/videos/show.html.haml

```diff
@@ -11,7 +11,7 @@
         .actions
           %a.btn.btn-primary(href="") Watch Now
           .inline
-            = link_to '+ My Queue', queue_items_path(video_id: @video.id), method: :post, class: 'btn btn-default'
+            = link_to '+ My Queue', queue_items_path(video_id: @video.id), method: :post, class: 'btn btn-default' if !current_user.video_in_queue?(@video)
```

Comment:
really don't know why but I'm still tempted to write ```if!``` instead of ```unless```. every and each time...
fixed.